lib/ansible/modules/remote_management/foreman/foreman_global_parameter.py

```diff
@@ -0,0 +1,200 @@
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: foreman_global_parameter
+short_description: Manage Foreman Global Parameters
+description:
+    - Manage Foreman Global Parameter Entities
+version_added: "2.4"
+author: ["Bernhard Hopfenmueller(@Fobhep)",
+        "Matthias Dellweg (@mdellweg)"]
+requirements:
+    - "nailgun >= 0.29.0"
+    - "python >= 2.6"
+options:
+    server_url:
+        description:
+            - URL of Foreman server
+        required: true
+    username:
+        description:
+            - Username on Foreman server
+        required: true
+        default: true
+    password:
+        description:
+            - Password for user accessing Foreman server
+        required: true
+    verify_ssl:
+        description:
+            - Verify SSL of the Foreman server
+        required: false
+    name:
+        description:
+            - Name of the Global Parameter
+        required: true
+    value:
+        description:
+            - Value of the Global Parameter
+        required: false
+    state:
+        description:
+            - State of the Global Parameter
+        required: false
+        choices:
+            - present
+            - absent
+'''
+
+EXAMPLES = '''
+- name: "Create a Global Parameter"
+  local_action:
+      module: foreman_global_parameter
+      username: "admin"
+      password: "admin"
+      server_url: "https://fakeserver.com"
+      name: "TheAnswer"
+      value: "42"
+      state: present
+
+- name: "Delete a Global Parameter"
+  local_action:
+      module: foreman_global_parameter
+      username: "admin"
+      password: "admin"
+      server_url: "https://fakeserver.com"
+      name: "TheAnswer"
+      state: absent
+'''
+
+RETURN = '''# '''
+
+try:
+    from nailgun.config import ServerConfig
+    import nailgun.entity_mixins
+    import nailgun.entities
+    HAS_NAILGUN_PACKAGE = True
+except:
+    HAS_NAILGUN_PACKAGE = False
+
+from ansible.module_utils.basic import AnsibleModule
+
+'''CommonParameter needs to inherit functionalities
+   for searching, updating and deleting'''
+
+
+class CommonParameter(
+    nailgun.entities.CommonParameter,
+    nailgun.entity_mixins.Entity,
+    nailgun.entity_mixins.EntityCreateMixin,
+    nailgun.entity_mixins.EntityDeleteMixin,
+    nailgun.entity_mixins.EntityReadMixin,
+    nailgun.entity_mixins.EntitySearchMixin,
+    nailgun.entity_mixins.EntityUpdateMixin
+):
+
+    pass
+
+
+def create_CommonParameter(name, value):
```

Comment:
This mixture of snakecase and camcelcase for method names seems a bit strange and divergent from typical python best practices. A bit of syntactic nitpick but it is nice to follow standard practice.